Common/ipcchannel.h

```diff
@@ -36,6 +36,7 @@
 
 #include <boost/interprocess/managed_shared_memory.hpp>
 #include <boost/container/string.hpp>
+#include <functional>
```

Comment:
Why add this?

Reply:
It will not compile otherwise since std::function<void()> is used.
Not sure why it compiles for you without it honestly.